service/src/tests/mocks/parentchain_api_mock.rs

```diff
@@ -0,0 +1,76 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use itp_api_client_extensions::{ApiResult, ChainApi, StorageProof};
+use itp_test::builders::{
+	parentchain_block_builder::ParentchainBlockBuilder,
+	parentchain_header_builder::ParentchainHeaderBuilder,
+};
+use itp_types::{SignedBlock, H256};
+use sp_finality_grandpa::AuthorityList;
+
+pub struct ParentchainApiMock {
+	parentchain: Vec<SignedBlock>,
+}
+
+impl ParentchainApiMock {
+	pub(crate) fn new() -> Self {
+		ParentchainApiMock { parentchain: Vec::new() }
+	}
+
+	/// Initializes parentchain with a default block chain of a given length.
+	pub fn with_default_blocks(mut self, number_of_blocks: u32) -> Self {
+		self.parentchain = (1..=number_of_blocks)
+			.map(|n| {
+				let header = ParentchainHeaderBuilder::default().with_number(n).build();
+				let block = ParentchainBlockBuilder::default().with_header(header).build_signed();
+				block
+			})
+			.collect();
+		self
+	}
+}
+
+impl ChainApi for ParentchainApiMock {
+	fn last_finalized_block(&self) -> ApiResult<Option<SignedBlock>> {
+		ApiResult::Ok(self.parentchain.last().cloned())
+	}
+
+	fn signed_block(&self, _hash: Option<H256>) -> ApiResult<Option<SignedBlock>> {
+		todo!()
+	}
+
+	fn get_blocks(&self, from: u32, to: u32) -> ApiResult<Vec<SignedBlock>> {
+		let num_elements = to.checked_sub(from).map(|n| n + 1).unwrap_or(0);
```

Comment:
If we enforce `from' must be <= 'to'` in the real one, we might want to have it here too.

Reply:
We can do that, but I think we don't have to. The purpose of a mock should primarily be to serve the needs of the test, not to mirror the real implementation. If we want to test that an error is returned, if  `from > to`, then yes, the mock should do that (with a mocking framework, we would set that condition ad-hoc).